pkg/kubectl/resource_printer.go

```diff
@@ -599,6 +615,11 @@ func printPod(pod *api.Pod, w io.Writer, options PrintOptions) error {
 func printPodBase(pod *api.Pod, w io.Writer, options PrintOptions) error {
 	name := pod.Name
 	namespace := pod.Namespace
+	kind := options.KindName
```

Comment:
It looks like version/group/kind info gets stripped out of the object before its passed in here. So disregard my suggestion, the way you are currently getting kind is how it has to be done.